### ISSUE TITLE
Nats server memory usage slowly increases over time

### ISSUE THREAD

**Issue by Kaarel** [nats-io/nats-server#5670]
### Observed behavior

I haven't got much more than INFO level nats logs and RAM usage at this stage. The memory leak is slow enough that it never gets to a point where it hurts because we upgrade to new versions as they are released. But wanted to bring your attention to this anyway as the memory profile looks of concern. Or maybe this is expected until an system memory limit is reached?

![image](https://github.com/user-attachments/assets/1e744d2c-1b7e-445f-8509-e0499b310c82)

We have a 3 node cluster. Mostly JS and .net clients. Some PHP. All WorkQueue or Interest streams. Couple small KV_buckets. There is traffic but messages from streams are consumed immediately so most of the time empty.

My primary concern is the general steady memory increase.

For whatever reason the last drop on one of the nodes around 07/02 mark coincides with creating and deleting a temp stream for testing purposes. Nodes n1 and n2 had no server logs for that time period. Node n3 had just these two lines in its server logs:
```
[3750] 2024/07/02 14:33:51.591074 [INF] JetStream cluster new stream leader for 'USERS > test'
[3750] 2024/07/02 14:45:30.614779 [INF] JetStream cluster new stream leader for 'USERS > test1'
```
Also notably maybe, the drop in memory usage occurred not on the node with these log lines but on another node...

If there is any other info I could provide let me know.

### Expected behavior

I guess I would naively expect the memory consumption to plateau at some point. Something like what one of the cluster nodes demonstrates during the period of 2024/07/02 to 2024/07/12 (see attached screenshot). But that too started to grow after 07/12.

### Server and client version

2.10.14 and 2.10.16
See attached screenshot for details.

**Comment by neilalexander** [comment]
How are you reporting memory usage? Is this including buffers/caches or is this the resident set size of the NATS Server process?

Please fetch memory profiles from your NATS system using `nats server request profile allocs` under the system account.

**Comment by Kaarel** [comment]
The stats on diagram are `container_memory_usage_bytes` which as per https://prometheus.io/docs/guides/cadvisor/ is "The cgroup's total memory usage"

This is how we run NATS `docker run --network host --pid host nats`

Do you have a Synadia contact (slack/email) to use for profiles file.

**Comment by neilalexander** [comment]
The `container_memory_usage_bytes` will include the kernel buffers/caches. Can you please check whether the `container_memory_working_set_bytes` shows the same pattern?

Feel free to send profiles directly to me at neil@nats.io.

**Comment by Kaarel** [comment]
Yes the `container_memory_working_set_bytes` mirrors `container_memory_usage_bytes`. On the screenshot the sharp drop is where we restarted nodes.

![image](https://github.com/user-attachments/assets/6ba4658c-32d5-4a4f-b603-7bfa707f5fe1)

**Comment by NevinDry** [comment]
Hello,
It looks like we are observing the same issue on our NATS implementation. The memory slowly increase overtime.
What is weird is we can observe this behavior on our STG1 environement but not on our DEV environment. 
These two environments have no significant activity, and are deployed the same way through IaC. 
Nats is deployed using clustering on AWS fargate. 

The difference in memory usage between the two envs is very significant.
Let me show you some screen of one month of activity (we don't have traffic on these environments) :

dev:
![image](https://github.com/user-attachments/assets/0f145152-9780-4520-b569-bf1c8ccf71e9)

stg1:
![image](https://github.com/user-attachments/assets/c9c9f661-3dc1-4843-a4de-f05c7ebc0f16)

I was able to connect to NATS monitoring endpoints and I saw no significant difference between our dev and stg1.

Do you have any clue on what could cause this memory increase ?

We also have this behavior in our production environement that is why it is kind of problematic. 

Thank you very much.

Any help apreciated 👍

**Comment by wallyqs** [comment]
Looks like a duplicate of #5870
